src/middleware.ts

```diff
@@ -1,102 +1,85 @@
 import { AzureFunction, Context } from '@azure/functions';
-import { ApplicationError } from './applicationError';
 
-type ErrorWithMessage = {
-    message: string;
-    stack?: string;
-};
-
-const isErrorWithMessage = (error: unknown): error is ErrorWithMessage => {
-    return (
-        typeof error === 'object' &&
-        error !== null &&
-        'message' in error &&
-        typeof (error as Record<string, unknown>).message === 'string'
-    );
-};
+import { errorHandler } from './error';
 
-const logErrorObject = (error: object | null, context: Context) => {
-    if (error === null) {
-        context.log.error('The provided error was eq to null - unable to log a specific error-message');
-        return;
-    }
+type ErrorResult = { $failed: true; $error: unknown };
 
-    if (isErrorWithMessage(error)) {
-        context.log.error({ message: error.message, stack: error.stack });
-    } else {
-        try {
-            const errorAsJson = JSON.stringify(error);
-            if (errorAsJson === '{}') {
-                context.log.error(error.toString());
-            }
-        } catch (_) {
-            //Fallback in case there's an error stringify
-            context.log.error(error.toString());
-        }
-    }
-};
+const isErrorResult = (result: unknown | ErrorResult): result is ErrorResult => (result as ErrorResult)?.$failed;
 
 const middlewareCore =
-    (beforeExecution: AzureFunction[], handler: AzureFunction, afterExecution: AzureFunction[]) =>
-    async (context: Context, ...args: unknown[]): Promise<void> => {
+    (beforeExecution: AzureFunction[], handler: AzureFunction, postExecution: AzureFunction[]) =>
+    async (context: Context, ...args: unknown[]): Promise<unknown | ErrorResult> => {
         if (beforeExecution) {
             for (const middlewareFunctions of beforeExecution) {
                 await middlewareFunctions(context, ...args);
             }
         }
-        const handlerResult = await handler(context, ...args);
-        if (afterExecution) {
-            for (const middleware of afterExecution) {
+
+        let handlerResult;
+        let error = undefined;
+
+        try {
+            handlerResult = await handler(context, ...args);
+        } catch (err) {
+            error = err;
+        }
+
+        // TODO: Give post-execution functions a parameter to indicate if the handler failed.
```

Comment:
changes for a future PR ?

Reply:
Would be my suggestion to keep the PR somewhat smallish.